rasa/core/actions/forms.py

```diff
@@ -584,6 +585,28 @@ async def activate(
         # collect values of required slots filled before activation
         prefilled_slots = {}
 
+        # run slot extraction the second time during form activation
+        # for slots with mapping conditions
+        action_extract_slots = action.action_for_name_or_text(
+            ACTION_EXTRACT_SLOTS, domain, self.action_endpoint
+        )
+
+        logger.debug(
+            f"Executing default action '{ACTION_EXTRACT_SLOTS}' at form activation."
+        )
+
+        extraction_events = await action_extract_slots.run(
+            output_channel, nlg, tracker, domain
+        )
+
+        events_as_str = "\n".join([str(e) for e in extraction_events])
+        logger.debug(
+            f"The execution of '{ACTION_EXTRACT_SLOTS}' resulted in "
+            f"these events: {events_as_str}."
+        )
+
+        tracker.update_with_events(extraction_events, domain)
```

Comment:
I have to update the tracker immediately in order for the tracker to update its `tracker.slots` attribute, which gets used in the for loop below (L610-612) to extract the prefilled slots. Because the tracker is temporary, it won't duplicate the events in the **actual** tracker that the processor updates.